File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        low = 0 
        high = len(nums) - 2
        pivot = high + 1
        while ( low <= high ):
            mid = low + (high - low) // 2
            if (nums[mid] > nums[mid + 1]):
                pivot = mid
                break
            elif nums[mid] > nums[pivot]:
                low = mid + 1
            else:
                high = mid - 1

        return nums[(pivot + 1) % len(nums)]

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int n = nums.size();
         int achor = nums[0];
         int low = 1, high = n-1;
         while(low <= high) {
            int mid = low + (high - low)/2;
            if(nums[mid] < nums[mid-1]) {
                return nums[mid];
            } else if(nums[mid] > achor){
                low = mid + 1;
            } else {
                high = mid - 1;
            }
         }
         return achor;
    }
};